sdc-plugin/clocks.cc

```diff
@@ -226,8 +229,11 @@ void Clock::UpdatePeriod(float period) {
 void Clock::UpdateWaveform(float rising_edge, float falling_edge) {
     rising_edge_ = rising_edge;
     falling_edge_ = falling_edge;
-    if (falling_edge_ > period_) {
-	log_error("Phase shift on clock %s exceeds 360 degrees\nRising edge: %f, Falling edge: %f, Clock period:%f\n", name_.c_str(), rising_edge_, falling_edge_, period_);
+    if (rising_edge_ >= period_ ) {
```

Comment:
If `ApplyShift` is called with a rising_edge greater than the period (which is legal), just subtracting 1 period isn't enough.

Also `ApplyShift` allows being invoked with a negative rising edge in the current code.  Maybe add an error if ApplyShift is called with a negative value?  If not, then add code to handle falling/rising edge < 0?

Reply:
Fixed by using `fmod` for period and waveform calculations.